projects/packages/forms/src/blocks/contact-form/components/jetpack-field-number.js

```diff
@@ -0,0 +1,153 @@
+import { useBlockProps } from '@wordpress/block-editor';
+import { createBlock, getDefaultBlockName } from '@wordpress/blocks';
+import { __experimentalNumberControl as NumberControl } from '@wordpress/components'; // eslint-disable-line @wordpress/no-unsafe-wp-apis
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import { isEmpty } from 'lodash';
+import { useFormStyle } from '../util/form';
+import { withSharedFieldAttributes } from '../util/with-shared-field-attributes';
+import JetpackFieldControls from './jetpack-field-controls';
+import JetpackFieldLabel from './jetpack-field-label';
+import { useJetpackFieldStyles } from './use-jetpack-field-styles';
+
+const JetpackNumberField = props => {
+	const {
+		attributes,
+		clientId,
+		id,
+		isSelected,
+		required,
+		requiredText,
+		label,
+		setAttributes,
+		placeholder,
+		min,
+		max,
+		width,
+		insertBlocksAfter,
+	} = props;
+
+	const { blockStyle, fieldStyle } = useJetpackFieldStyles( attributes );
+	const formStyle = useFormStyle( clientId );
+	const blockProps = useBlockProps( {
+		className: clsx( 'jetpack-field', {
+			'is-selected': isSelected,
+			'has-placeholder': ! isEmpty( placeholder ),
+		} ),
+		style: blockStyle,
+	} );
+
+	return (
+		<>
+			<div { ...blockProps }>
+				<JetpackFieldLabel
+					attributes={ attributes }
+					label={ label }
+					required={ required }
+					requiredText={ requiredText }
+					setAttributes={ setAttributes }
+					style={ formStyle }
+				/>
+				<input
+					className="jetpack-field__input"
+					onChange={ e => setAttributes( { placeholder: e.target.value } ) }
+					style={ fieldStyle }
+					type="number"
+					value={ placeholder }
+					min={ min }
+					max={ max }
```

Comment:
These min/max are also a difference from `jetpack-field.js`. And fixed `type="number"` but that was non issue already before.